components/usage/pkg/apiv1/billing.go

```diff
@@ -100,7 +106,7 @@ func (s *BillingService) GetStripeCustomer(ctx context.Context, req *v1.GetStrip
 			if errors.Is(err, db.ErrorNotFound) {
 				stripeCustomer, err := s.stripeClient.GetCustomer(ctx, identifier.StripeCustomerId)
 				if err != nil {
-					return nil, err
+					return nil, status.Errorf(codes.NotFound, "Failed to find customer with ID %s: %s", identifier.StripeCustomerId, err.Error())
```

Comment:
Shouldn't be needed, `stripeClient.GetCustomer` already returns a this NotFound error